pkg/kv/kvtest/store.go

```diff
@@ -0,0 +1,269 @@
+package kvtest
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type makeStore func(t *testing.T, ctx context.Context) kv.Store
+
+func TestDriver(t *testing.T, name string, dsn string) {
+	ms := makeStoreByName(name, dsn)
+	t.Run("Driver_Open", func(t *testing.T) { testDriverOpen(t, ms) })
+	t.Run("Store_SetGet", func(t *testing.T) { testStoreSetGet(t, ms) })
+	t.Run("Store_SetIf", func(t *testing.T) { testStoreSetIf(t, ms) })
+	t.Run("Store_Delete", func(t *testing.T) { testStoreDelete(t, ms) })
+	t.Run("Store_Scan", func(t *testing.T) { testStoreScan(t, ms) })
+}
+
+func testDriverOpen(t *testing.T, ms makeStore) {
+	ctx := context.Background()
+	store1 := ms(t, ctx)
+	store1.Close()
+
+	store2 := ms(t, ctx)
+	store2.Close()
+}
+
+func testStoreSetGet(t *testing.T, ms makeStore) {
+	ctx := context.Background()
+	store := ms(t, ctx)
+	defer store.Close()
+
+	testKey := []byte("key")
+	testValue1 := []byte("value1")
+	testValue2 := []byte("value2")
+
+	// set test key with value1
+	err := store.Set(ctx, testKey, testValue1)
+	if err != nil {
+		t.Fatalf("failed to set key '%s', to value '%s': %s", testKey, testValue1, err)
```

Comment:
In this specific case, if calling 'Set' fails, means that calling 'Get' will probably fail too - not getting a list of errors, the first one is enough in this case.
For single operation which multiple tests, I usually user error to check - ex: Add repository, get repository details and check each expected field value and print error - or use one diff with fatal ;)

Reply:
I think we've had this conversation a million times: I like to fail only when the test will crash if it continues; otherwise I like to error out and get ONE MILLION error lines in the log.

[![Dr Evil: one MILLION errors!](https://user-images.githubusercontent.com/7802932/165085977-9f00b408-46f1-45f6-8593-0d7a6c2f6ec8.png)](https://imgflip.com/i/6drbbq).
I think I should stop repeating this comment, I doubt we will ever agree.